server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/StructureComparisonContext.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.structurecompare;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.validation.constraints.NotNull;
+
+import com.oceanbase.odc.service.flow.task.model.DBStructureComparisonTaskResult;
+
+import lombok.NonNull;
+
+/**
+ * @author jingtian
+ * @date 2024/1/18
+ * @since ODC_release_4.2.4
+ */
+public class StructureComparisonContext implements Future<DBStructureComparisonTaskResult> {
+    private StructureComparisonTask task;
+    private Future<DBStructureComparisonTaskResult> controlFuture;
+
+    public StructureComparisonContext(@NonNull StructureComparisonTask task,
+            @NonNull Future<DBStructureComparisonTaskResult> controlFuture) {
+        this.task = task;
+        this.controlFuture = controlFuture;
+    }
+
+    public double getProgress() {
+        return task.getProgress();
+    }
+
+    public DBStructureComparisonTaskResult getStatus() {
```

Comment:
why will we get a result entity by calling this 'status' method?

Reply:
Because this method is to obtain the task status during the running of the structure comparison thread, if we want to obtain the thread result, use `context.get()` directly.